app/models/login.rb

```diff
@@ -33,9 +33,9 @@ class InvalidOAuth2Token < StandardError; end
     has_secure_password
   end
 
-  validates :identification, presence: true
-  validates :oauth2_token, presence: true
-  validates :single_use_oauth2_token, presence: true
+  validates :identification, presence: true, uniqueness: true
+  validates :oauth2_token, presence: true, uniqueness: true
+  validates :single_use_oauth2_token, presence: true, uniqueness: true
```

Comment:
The only way to test that I could think of then is to `stub` the `ensure_oauth2_token` and `assign_single_use_oauth2_token` methods on the instance then to return a static value then. Stubbing private API isn't great but still better than not testing this at all.